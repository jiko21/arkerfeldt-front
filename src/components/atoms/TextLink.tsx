/** @jsxImportSource @emotion/react */
import Link from 'next/link';
import { css } from '@emotion/react';
import { Color } from '@/const/color';
import * as React from 'react';

const hrefStyle = (fontSize: number) => css`
  color: ${Color.ENABLE};
  cursor: pointer;
  font-size: ${fontSize}px;
  font-weight: 600;
`;

interface IProps {
  href: string;
  fontSize: number;
}

const TextLink: React.FC<IProps> = ({ children, href, fontSize }) => (
  <Link href={href}>
    <a css={hrefStyle(fontSize)}>{children}</a>
  </Link>
);

export default TextLink;
