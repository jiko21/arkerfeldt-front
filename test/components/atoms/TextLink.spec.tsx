import TextLink from '@/components/atoms/TextLink';
import { render } from '@testing-library/react';

describe('TextLink.tsx', () => {
  let onClickMock: jest.Mock;
  beforeEach(() => {
    onClickMock = jest.fn();
  });

  afterEach(() => {
    onClickMock.mockClear();
  });
  it('should correctly render', () => {
    const { container } = render(
      <TextLink href="/" fontSize={20}>
        Sample
      </TextLink>,
    );

    expect(container).toMatchSnapshot();
  });
});
