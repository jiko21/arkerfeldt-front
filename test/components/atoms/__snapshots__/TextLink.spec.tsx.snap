// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLink.tsx should correctly render 1`] = `
.emotion-0 {
  color: #007AAE;
  cursor: pointer;
  font-size: 20px;
  font-weight: 600;
}

<div>
  <a
    class="emotion-0"
  >
    Sample
  </a>
</div>
`;
